src/main/scala/is/hail/expr/ir/functions/MathFunctions.scala

```diff
@@ -79,5 +80,21 @@ object MathFunctions extends RegistryFunctions {
     registerJavaStaticFunction("%", TInt64(), TInt32(), TInt64())(jMathClass, "floorMod")
 
     registerScalaFunction("isnan", TFloat64(), TBoolean())(thisClass, "isnan")
+
+    registerCodeWithMissingness("&&", TBoolean(), TBoolean(), TBoolean()) { (_, l, r) =>
+      EmitTriplet(
+        Code(l.setup, r.setup),
+        l.m || (coerce[Boolean](l.v) && r.m),
+        coerce[Boolean](l.v) && coerce[Boolean](r.v)
+      )
+    }
+
+    registerCodeWithMissingness("||", TBoolean(), TBoolean(), TBoolean()) { (_, l, r) =>
+      EmitTriplet(
+        Code(l.setup, r.setup),
+        l.m || (!coerce[Boolean](l.v) && r.m),
+        coerce[Boolean](l.v) || coerce[Boolean](r.v)
```

Comment:
is there a difference between `coerce` and `asInstanceOf`?

Reply:
`coerce[T](code)` is a utility function that Dan wrote to have a more concise way of writing `code.asInstanceOf[Code[T]]` everywhere.